packages/ef-runtime-client/src/ui/efui.tsx

```diff
@@ -0,0 +1,126 @@
+import { Component } from "preact/compat";
+import { IComponentRegistry } from "../ComponentRegistry";
+import styles from "./styles";
+
+type EFUIProps = { componentRegistry: IComponentRegistry };
+type EFUIState = {
+  showComponents: boolean;
+  needsRefresh: boolean;
+  components: { [key: string]: string };
+};
```

Comment:
```suggestion
interface ComponentInfo {
  [key: string]: string;
}

type EFUIState = {
  showComponents: boolean;
  needsRefresh: boolean;
  components: ComponentInfo;
};
```

Maybe add an interface here?